### PR TITLE
Conversation feature

### PR DESCRIPTION
**Pull request by ghost**
## Conversation feature

PT: https://www.pivotaltracker.com/story/show/160646357

Add feature test to: 
- Send a message from one user to another user. 
- Receive a message from the previous user and count how many messages in the mailbox.
- Delete that message to the trash box.

### CODE REVIEW

File: features/conversation.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        Given I should be logged in as "Tore"
+        And I am on the landing page
+        And I send a mail to "Olof"
+        And I click on the "Logout" button
+        Given I should be logged in as "Olof"
+        And I am on the landing page
+        And I click on the "Inbox" button
+        Then I should have "1" new messages
+        And I click on the "View" button
+        And I click on the "Move to trash" button
```
Comment (oliverochman) [inline features/conversation.feature]: Out of scope for this feature

File: features/conversation.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    I should be able to send and receive message from others
+
+    Background: User is Logged in
+        Given the following user is in database
+        | email              | password   | name |
+        | tore@example.com   | foobarbar  | Tore |
+        | olof@example.com   | mypassword | Olof |
+    
+    Scenario: As a user I should be able to send and receive messages to and from others
+        Given I should be logged in as "Tore"
```
Comment (oliverochman) [inline features/conversation.feature]: Line 13 to 18 can be in the background.

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/conversation.feature [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    As a user 
+    in order to have a conversation with other people 
+    I should be able to send and receive message from others
+
+    Background: User is Logged in and sends message
+        Given the following user is in database
+        | email              | password   | name |
+        | tore@example.com   | foobarbar  | Tore |
+        | olof@example.com   | mypassword | Olof |
+        Given I should be logged in as "Tore"
```
Comment (magnus-thor) [inline features/conversation.feature]: You can refactor the next 4 steps and do all of this with a table
```
Given these mails exists
| sender | receiver | subject | body  |
| Tore     | Olof       | Hello    | World |
```
Then in the step definition you use the table information to send the email

**Review by magnus-thor — CHANGES_REQUESTED** [review]

**Review by magnus-thor — APPROVED** [review]
Looks good

**Pull request merged**